### PR TITLE
Update install texts

### PR DESCRIPTION
**Pull request by glennsong09**
Update INSTALL.txt to INSTALL_Autotools.txt and update options in INSTALL_CMake.txt as well as update some Cmake text to be consistent with the rest.

### CODE REVIEW

File: release_docs/INSTALL_Autotools.txt
Comment (byrnHDF) [inline release_docs/INSTALL_Autotools.txt:1]: I have some changes to this file along the same lines as I did in HDF4.
The other files are good.

**Review by byrnHDF — APPROVED** [review]

**Review by hyoklee — APPROVED** [review]

File: release_docs/INSTALL_Autotools.txt
Comment (byrnHDF) [reply release_docs/INSTALL_Autotools.txt:1]: See #4013

Comment (byrnHDF) [reply release_docs/INSTALL_Autotools.txt:1]: I can move your GNU_DIRS changes to HDF4.

File: config/cmake/HDFMacros.cmake [hunk 1]
```diff
@@ -369,7 +369,7 @@ macro (HDFTEST_COPY_FILE src dest target)
 endmacro ()
 
 macro (HDF_DIR_PATHS package_prefix)
-  option (HDF5_USE_GNU_DIRS "TRUE to use GNU Coding Standard install directory variables, FALSE to use historical settings" FALSE)
+  option (HDF5_USE_GNU_DIRS "ON to use GNU Coding Standard install directory variables, OFF to use historical settings" OFF)
```
Comment (byrnHDF) [inline config/cmake/HDFMacros.cmake:372]: There is an equivalent option in ExamplesCache.cmake file.

Comment (byrnHDF) [reply config/cmake/HDFMacros.cmake:372]: And there is an HDFMacros.cmake file in HDF5Examples/config/cmake folder

Comment (glennsong09) [reply config/cmake/HDFMacros.cmake:372]: I have updated these two files.

**Review by byrnHDF — APPROVED** [review]

**Comment by byrnHDF** [comment]
> Update INSTALL.txt to INSTALL_Autotools.txt and update options in INSTALL_CMake.txt as well as update some Cmake text to be consistent with the rest.

INSTALL_Autotools.txt to INSTALL_Auto.txt (just because I already did that in HDF4) ;{)

**Comment by glennsong09** [comment]
> > Update INSTALL.txt to INSTALL_Autotools.txt and update options in INSTALL_CMake.txt as well as update some Cmake text to be consistent with the rest.
> 
> INSTALL_Autotools.txt to INSTALL_Auto.txt (just because I already did that in HDF4) ;{)

I just updated that to match your changes.

**Review by lrknox — APPROVED** [review]

**Pull request merged**